V.2.0.0 proofreading changes (#128)
uptane/deployment-considerations: exceptional_operations.md

### Aftermarket ECUs

A slightly more difficult use case to deal with concerns the use of aftermarket ECUs -- for example, 3rd-party replacement parts, or add-on ECUs that add functionality for commercial fleet management. Though from a technical perspective adding an aftermarket ECU can be managed in one of the ways recommended in the previous subsection, there is no doubt that these components bring with them a set of unique logistical and security concerns. For starters, because aftermarket suppliers may not have access to the original design, these ECUs are often products of reverse engineering. As such, suppliers may not be able to glean all relevant design information about the ECU, including the rationale behind certain choices. Furthermore, the use of these components raises a number of fundamental questions, such as:
- If an aftermarket ECU does not have its own primary, will it still be able to get updates through an existing OEM primary ECU, as long as the  OEMs Director repository permits it? 
- If an aftermarket ECU does have its own Primary, is each capable of controlling a mutually exclusive set of Secondaries?

> - Could an owner (or a third-party) direct updates for its own vehicle from both an OEM and an aftermarket source? 

Author: ```suggestion
- Could owners (or third-parties) direct updates for their own vehicle from both an OEM and an aftermarket source? 
```

> - Could an owner (or a third-party) direct updates for its own vehicle from both an OEM and an aftermarket source? 

Author: I made it plural. Hence, no problem. You are right that people should never be "its," but I wasn't sure if the "third parties" here were people or some type of automated machinery.

To safely remove a tier-1 supplier, the OEM SHOULD use the following steps. First, it SHOULD delete the corresponding delegation from the Targets role on the Image repository, as well as all metadata and images belonging to that supplier, so that their metadata and images are no longer trusted. Second, it SHOULD also delete information about the supplier from the Director repository, such as its images, as well as its dependencies and conflicts, so that the Director repository no longer chooses these images for installation. In order to continue to update vehicles with ECUs originally maintained by this supplier, the OEM SHOULD replace this supplier with another delegation, either one maintained by the OEM, or by another tier-1 supplier.

> Note that to comply with the [Standard](https://uptane.github.io/papers/uptane-standard.1.1.0.html#check_snapshot), the Snapshot metadata must continue to list the removed delegation in order to prevent a rollback attack. However, if the OEM rotates the Timestamp and Snapshot keys (and pushes new Root metadata with the new keys), the delegation may be safely removed from the Snapshot metadata. As the ECU will need to clear out any existing Snapshot metadata due to the rotation, checking to see that each Targets metadata filename listed in the previous Snapshot metadata is also listed in the new Snapshot metadata will (trivially) not apply.

Reviewer: This is another change I don't get. "[T]he check [...] will [...] not apply" reads fine to me. "[C]hecking [...] will [...] not be applicable" is the other option, but that sounds like a downgrade to me.

Author: I'll go with your instincts here. I just thought it read better the other way.
```suggestion
Note that to comply with the [Standard](https://uptane.github.io/papers/uptane-standard.1.1.0.html#check_snapshot), the Snapshot metadata must continue to list the removed delegation in order to prevent a rollback attack. However, if the OEM rotates the Timestamp and Snapshot keys (and pushes new Root metadata with the new keys), the delegation may be safely removed from the Snapshot metadata. As the ECU will need to clear out any existing Snapshot metadata due to the rotation, the check that each Targets metadata filename listed in the previous Snapshot metadata is also listed in the new Snapshot metadata will (trivially) not apply.
```
